### PR TITLE
Stargate prometheus

### PR DESCRIPTION
**Pull request by polandll**
Added an example docker-compose yaml file for stargate + prometheus + grafana. A shell script is included to run the docker-compose, to avoid timing issues. A cqlsh command to start cqlsh within the container is also included.

### CODE REVIEW

File: examples/stargate-prometheus-grafana/cqlsh-commands.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+backend-1
+docker run --rm -it nuvo/docker-cqlsh cqlsh host.docker.internal 9044 --cqlversion=3.4.4
```
Comment (dougwettlaufer) [inline examples/stargate-prometheus-grafana/cqlsh-commands.txt:2]: We need to set cqlversion on these?

File: examples/stargate-prometheus-grafana/start_stargate-prometheus-grafana.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#!/bin/sh
+
+COMPOSE_FILE=stargate-prometheus-grafana.yml
+
+# Make sure backend-1, the seed node, is up before bringing up other nodes and stargate
+docker-compose --file $COMPOSE_FILE up -d backend-1
+sleep 60
```
Comment (dougwettlaufer) [inline examples/stargate-prometheus-grafana/start_stargate-prometheus-grafana.sh:7]: Something I recently discovered 

```sh
(docker-compose logs -f backend-1 &) | grep -q "DSE startup complete."
```

That will cause the script to wait while tailing the logs until the "DSE startup complete" message is logged.

File: examples/stargate-prometheus-grafana/start_stargate-prometheus-grafana.sh [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+# Bring up the 2nd C* node
+#docker-compose --file $COMPOSE_FILE up -d backend-2
+#sleep 60
+# Bring up the 3rd C* node
+#docker-compose --file $COMPOSE_FILE up -d backend-3
+#sleep 60
+# Bring up the stargate
+docker-compose --file $COMPOSE_FILE up -d stargate
+sleep 60
```
Comment (dougwettlaufer) [inline examples/stargate-prometheus-grafana/start_stargate-prometheus-grafana.sh:16]: Likewise you can do something like this to wait for stargate

```sh
echo ""
echo "Waiting for stargate to start up..."
while [[ "$(curl -s -o /dev/null -w ''%{http_code}'' http://localhost:8082/health)" != "200" ]]; do
    printf '.' 
    sleep 5
done
```

File: examples/stargate-prometheus-grafana/start_stargate-prometheus-grafana.sh [hunk 4]
```diff
@@ -0,0 +1,18 @@
+#!/bin/sh
+
+COMPOSE_FILE=stargate-prometheus-grafana.yml
```
Comment (dougwettlaufer) [inline examples/stargate-prometheus-grafana/start_stargate-prometheus-grafana.sh:3]: Just FYI, naming the file "docker-compose.yaml" means you don't have to provide it in commands

**Review by dougwettlaufer — DISMISSED** [review]

File: examples/stargate-prometheus-grafana/cqlsh-commands.txt (hunk 1 above)
Comment (polandll) [reply examples/stargate-prometheus-grafana/cqlsh-commands.txt:2]: Yes, I found that I had to specify or it failed.

File: examples/stargate-prometheus-grafana/start_stargate-prometheus-grafana.sh (hunk 2 above)
Comment (polandll) [reply examples/stargate-prometheus-grafana/start_stargate-prometheus-grafana.sh]: I changed this, but used a C* log message `now part of the of the cluster`.

File: examples/stargate-prometheus-grafana/start_stargate-prometheus-grafana.sh (hunk 4 above)
Comment (polandll) [reply examples/stargate-prometheus-grafana/start_stargate-prometheus-grafana.sh]: I changed the file to the default "docker-compose.yml".

File: examples/stargate-prometheus-grafana/start_stargate-prometheus-grafana.sh (hunk 3 above)
Comment (polandll) [reply examples/stargate-prometheus-grafana/start_stargate-prometheus-grafana.sh]: I incorporated this wait logic.

File: examples/stargate-prometheus-grafana/start_stargate-prometheus-grafana.sh [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+# Make sure backend-1, the seed node, is up before bringing up other nodes and stargate
+docker-compose up -d backend-1
+# Wait until the seed node is up before bringing up more nodes
+(docker-compose logs -f backend-1 &) | grep -q "is now part of the cluster"
+
+# Bring up the 2nd C* node
+(docker-compose logs -f backend-1 &) | grep -q "is now part of the cluster"
+# Bring up the 3rd C* node
+(docker-compose logs -f backend-1 &) | grep -q "is now part of the cluster"
```
Comment (dougwettlaufer) [inline examples/stargate-prometheus-grafana/start_stargate-prometheus-grafana.sh:13]: Should these have corresponding `docker-compose up -d backend-2` and `docker-compose up -d backend-3`?

Comment (polandll) [reply examples/stargate-prometheus-grafana/start_stargate-prometheus-grafana.sh]: Yes, added those.

**Review by dougwettlaufer — APPROVED** [review]

**Pull request merged**